scripts/normalize-by-median.py

```diff
@@ -43,23 +45,13 @@ def batchwise(coll, size):
 # Returns true if the pair of records are properly pairs
 
 
-# pylint: disable=too-many-locals,too-many-branches
-def normalize_by_median(input_filename, outfp, htable, paired, cutoff,
-                        report_fp=None):
+def WithDiagnostics(ifile, batch_size, fp, paired):
+    index = 0
+    global total, discarded
 
-    desired_coverage = cutoff
-    ksize = htable.ksize()
+    for batch in batchwise(screed.open(ifile, parse_description=False),
+                           batch_size):
 
-    # In paired mode we read two records at a time
-    batch_size = 1
-    if paired:
-        batch_size = 2
-
-    index = -1
-    total = 0
-    discarded = 0
-    for index, batch in enumerate(batchwise(screed.open(
-            input_filename, parse_description=False), batch_size)):
         if index > 0 and index % 100000 == 0:
             print >>sys.stderr, '... kept {kept} of {total} or'\
```

Comment:
I think this 'print' could usefully be encapsulated in a function; just a thought.